### PR TITLE
Fix test refactoring

### PR DESCRIPTION
**Pull request by aaronliu0130**
Replace self.fail()s
Fix edge case of verifying categories when running pytest on a specific function through module notation Fix indented parameter test
Make _Seen_Error_Categories a set
Make clitest setup and teardown conform to naming

### CODE REVIEW

File: cpplint_unittest.py [hunk 1]
```diff
@@ -5259,9 +5257,7 @@ def testBuildHeaderGuard(self):
             ],
             error_collector,
         )
-        for line in error_collector.ResultList():
-            if line.find("build/header_guard") != -1:
-                self.fail("Unexpected error: %s" % line)
+        assert not any("build/header_guard" in line for line in error_collector.ResultList())
```
Comment (cclauss) [inline cpplint_unittest.py]: Although [`ruff rule SIM110`](https://docs.astral.sh/ruff/rules/reimplemented-builtin) recommends the use of `any()` because it is much easier to read and understand, it may make pytest results harder to understand because it may not show the text on those lines that fail.


Also, (I learned the hard way) that `any()` is actually slower than the long form that it replaces.

Comment (aaronliu0130) [reply cpplint_unittest.py]: Apparently it is faster than the form it replaces if you use it like `any(True for line in error_collector.ResultList() if "build/header_guard" in line)` instead, but you're right that it doesn't show the line.

Comment (aaronliu0130) [reply cpplint_unittest.py:5260]: Should I change them all to the early exit–less form `assert not "build/header_guard" in (line for line in error_collector.ResultList())`?

**Review by cclauss — APPROVED** [review]
